core/include/irods/private/http_api/transport.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef IRODS_HTTP_API_TRANSPORT_HPP
+#define IRODS_HTTP_API_TRANSPORT_HPP
+
+#include <string_view>
+#include <memory>
+
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl.hpp>
+#include <boost/beast/core/tcp_stream.hpp>
+#include <boost/beast/http/string_body.hpp>
+#include <boost/beast/http/read.hpp>
+#include <boost/beast/http/write.hpp>
+#include <boost/beast/ssl.hpp>
+#include <boost/url/parse.hpp>
+
+// Likely to be a pure virtual (interface)
```

Comment:
Should this comment be above `class transport`?

Some Doxygen could be good for these classes, too. I know it's early days, so, up to you guys.

Reply:
Comment could be thrown out, since it doesn't match up with the code. Maybe on Doxygen, @korydraughn any opinions?

Reply:
Let's remove it for now.

Reply:
Let's hold on doxygen for this PR.

Open an issue for that and phrase it like "Consider adding doxygen-based documentation" or something similar.

Reply:
Done

Reply:
if you make a new issue, please link it when you follow up so things are findable/clickable later.

in this case, looks like #230 is the new one.